src/components/motor_output_reordering/MotorOutputReorderingComponent.js

```diff
@@ -0,0 +1,249 @@
+            this._domSaveButton.show();
+        } else {
+            this._domStartOverButton.hide();
+            this._domSaveButton.hide();
+        }
+    }
+
+    _onAgreeButtonClicked() {
+        this._domActionHintBlock.text(i18n.getMessage("motorOutputReorderDialogSelectSpinningMotor"));
+        this._domWarningContentBlock.hide();
+        this._domMainContentBlock.show();
+        this.startUserInteraction();
+    }
+
+    _stopUserInteraction() {
+        if (this.motorOutputReorderCanvas) {
+            this.motorOutputReorderCanvas.pause();
+        }
+    }
+
+    startUserInteraction() {
+        if (this.motorOutputReorderCanvas) {
+            this.motorOutputReorderCanvas.startOver();
+        } else {
+            this.motorOutputReorderCanvas = new MotorOutputReorderCanvas(this._domCanvas,
+                this._droneConfiguration,
+                (motorIndex) => { // motor click callback
+                    this._onMotorClick(motorIndex);
+                },
+                (motorIndex) => { // motor spin callback
+                    let indexToSpin = -1;
+
+                    if (-1 !== motorIndex) {
+                        indexToSpin = this.motorOutputReorderCanvas.readyMotors.indexOf(motorIndex);
+                    }
+
+                    this._spinMotor(indexToSpin);
+                },
+            );
+        }
+
+        this._startMotorJerking(0);
+    }
+
+    _stopAnyMotorJerking() {
+        if (-1 !== this._currentJerkingTimeout) {
+            clearTimeout(this._currentJerkingTimeout);
+            this._currentJerkingTimeout = -1;
+            this._spinMotor(-1);
+        }
+
+        this._currentJerkingMotor = -1;
+    }
+
+    _startMotorJerking(motorIndex) {
+        this._stopAnyMotorJerking();
+        this._currentJerkingMotor = motorIndex;
+        this._motorStartTimeout(motorIndex);
+    }
+
+    _motorStartTimeout(motorIndex) {
+        this._spinMotor(motorIndex);
+        this._currentJerkingTimeout = setTimeout(() => {
+            this._motorStopTimeout(motorIndex);
+        }, 250);
+    }
+
+    _motorStopTimeout(motorIndex) {
+        this._spinMotor(-1);
+        this._currentJerkingTimeout = setTimeout(() => {
+            this._motorStartTimeout(motorIndex);
+        }, 500);
+    }
+
+
+    _spinMotor(motorIndex) {
+        this._currentSpinningMotor = motorIndex;
+        const buffer = [];
+
+        for (let  i = 0; i < this._config[this._droneConfiguration].Motors.length; i++) {
+            if (i === motorIndex) {
+                buffer.push16(this._motorSpinValue);
+            } else {
+                buffer.push16(this._motorStopValue);
+            }
+        }
+
+        MSP.send_message(MSPCodes.MSP_SET_MOTOR, buffer);
+    }
+
+    _stopMotor()
+    {
+        if (-1 !== this._currentSpinningMotor) {
+            this._spinMotor(-1);
+        }
+    }
+
+    _onMotorClick(motorIndex) {
+        console.log(motorIndex);
```

Comment:
If you are gonna use the console to log some messages, give the message some more texts. If not, it will only appear some random numbers in the console.

Reply:
Agree, This one is forgotten. Will remove for sure.